commands/store.go

```diff
@@ -33,14 +33,16 @@ const (
 func init() {
 	// Setup flags that are used by multiple commands (variables defined in faas.go)
 	storeCmd.Flags().StringVarP(&gateway, "gateway", "g", defaultGateway, "Gateway URL starting with http(s)://")
+	storeCmd.PersistentFlags().StringVarP(&storeAddress, "url", "u", defaultStore, "Alternative URL starting with http(s)://")
 
 	// Setup flags used by store command
-	storeListCmd.Flags().StringVarP(&storeAddress, "url", "u", defaultStore, "Alternative URL starting with http(s)://")
 	storeListCmd.Flags().BoolVarP(&verboseDescription, "verbose", "v", false, "Verbose output for the field values")
 
-	storeInspectCmd.Flags().StringVarP(&storeAddress, "url", "u", defaultStore, "Alternative Store URL starting with http(s)://")
 	storeInspectCmd.Flags().BoolVarP(&verboseDescription, "verbose", "v", false, "Verbose output for the field values")
 
+	// Setup flags that are used by multiple commands (variables defined in faas.go)
+	storeDeployCmd.Flags().StringVarP(&gateway, "gateway", "g", defaultGateway, "Gateway URL starting with http(s)://")
```

Comment:
Just for my knowledge, why deploy command needs this flag explicitly but for `list` and `inspect` the [parent command flag](https://github.com/openfaas/faas-cli/pull/309/files#diff-4d8c2dcb33807ea71d97dffb92481be6R35) is enough?

Reply:
The commands are heading off in different directions.  list and inspect are hitting GitHub (or a provided URI) for JSON file and printing out detail contained in that JSON file.  Deploy is providing detail into a gateway (remote if provided, localhost if not) for the application to act upon.

Reply:
You're right. I was thinking this PR tries to make `gateway` flag available for all sub-commands which was not needed for list and inspect.

Reply:
Plus, we can get rid of [gateway flag on store command](https://github.com/openfaas/faas-cli/pull/309/files#diff-4d8c2dcb33807ea71d97dffb92481be6R35). @itscaro